hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -280,117 +326,130 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
-                self.c_process_market_pair(market_pair, market_pair_to_active_orders[market_pair])
+                await self.process_market_pair(timestamp, market_pair, market_pair_to_active_orders[market_pair])
+
             # log conversion rates every 5 minutes
-            if self._last_conv_rates_logged + (60. * 5) < self._current_timestamp:
+            if self._last_conv_rates_logged + (60. * 5) < timestamp:
                 self.log_conversion_rates()
-                self._last_conv_rates_logged = self._current_timestamp
+                self._last_conv_rates_logged = timestamp
         finally:
             self._last_timestamp = timestamp
 
-    def has_active_taker_order(self, object market_pair):
-        cdef dict market_orders = self._sb_order_tracker.c_get_market_orders()
+    async def get_gateway_quotes(self):
+        for market_pair in self._market_pairs.values():
+            if self.is_gateway_market(market_pair.taker):
+                _, _, quote_rate, _, _, base_rate = self.get_taker_to_maker_conversion_rate()
+                order_amount = self._order_amount * base_rate
+                order_price = await market_pair.taker.market.get_order_price(market_pair.taker.trading_pair, True, order_amount)
+                self._last_taker_buy_price = order_price
+                order_price = await market_pair.taker.market.get_order_price(market_pair.taker.trading_pair, False, order_amount)
+                self._last_taker_sell_price = order_price
+
+    def ready_for_new_trades(self) -> bool:
+        """
+        Returns True if there is no outstanding unfilled order.
+        """
+        for market_pair in self._market_pairs.values():
+            if self._has_unhedged_market_fill:
+                return False
+            if len(self.market_info_to_active_orders.get(market_pair.taker, [])) > 0:
+                return False
+        return True
+
+    async def apply_gateway_transaction_cancel_interval(self):
+        gateway_connectors: List[GatewayEVMAMM] = []
+        for market_pair in self._market_pairs.values():
+            if self.is_gateway_market(market_pair.taker):
+                gateway_connectors.append(cast(GatewayEVMAMM, market_pair.taker.market))
+
+        for gateway in gateway_connectors:
+            await gateway.cancel_outdated_orders(self._gateway_transaction_cancel_interval)
+
+    def has_active_taker_order(self, market_pair: MarketTradingPairTuple):
+        market_orders = self._sb_order_tracker.get_market_orders()
         if len(market_orders.get(market_pair, {})) > 0:
             return True
-        cdef dict limit_orders = self._sb_order_tracker.c_get_limit_orders()
+        limit_orders = self._sb_order_tracker.get_limit_orders()
         limit_orders = limit_orders.get(market_pair, {})
         if len(limit_orders) > 0:
             if len(set(limit_orders.keys()).intersection(set(self._maker_order_ids))) > 0:
                 return True
         return False
```

Comment:
Not sure what you mean by checking only for market orders. Right after that it checks for limit orders too.
But indeed the method name doesn't make sense, since the `_sb_order_tracker ` holds both maker and taker orders.